### DISCUSSION TITLE
Login with password manager on Firefox for iOS not possible

### DISCUSSION THREAD

**Discussion by wendevlin** [directus/directus#11933] [Archive]
I want to login to directus with my Bitwarden password manager in Firefox.

But I am just able to login when I retype the password by hand.

In Safari it works as expected. As far as I know, Firefox for iOS is just built around a headless Safari, because Apple doesn't allow any other web engine. I am not able to debug it any way because I don´t have a macbook. 
Moreover I never have any issues with other services in Firefox.

### To Reproduce

Use Firefox for iOS and try to login with a password manager or apple keyring.

### What version of Directus are you using?

9.4.3

### What version of Node.js are you using?

I use directus in docker

### What database are you using?

SQLite created by directus

### What browser are you using?

Firefox for iPhone

### What operating system are you using?

iOS 15.2.1

### How are you deploying Directus?

docker-compose

**Comment by oreilles** [comment]
Please describe the *exact* steps you're following, and the *exact* issue you're facing so that we can try to replicate your issue.

**Comment by wendevlin** [comment]
Sorry I forgot to answer:

info: Directus v5.9.2 is hosted via docker and is using a SQLite DB

Steps:
- On an iPhone 13 Mini, open the Firefox browser and then go to the directus login page
- Hit the email field
- Keyboard opens with the option to fill in passwords.
- Click the passwords option on the keyboard
- Bitwarden password manager opens
- Authenticate at bitwarden and select directus account to fill in
- Bitwarden automatically fills in email and password
- Keyboard automatically opens again
- hit done at the keyboard
- hit sign in
- nothing happens
  - No action at all, I am still at the login page and email and password are filled out
  - there is no error message
  - The Login button seems to be disabled all the time and never recognize that there is data in the fields.

I am able to login if I manually enter one key of the password. This works only with the password not with the email. But since it is a password manager used, I don't know the password and would need to get it from the PW manager.

If I do the steps with Safari it works as expected. What is pretty strange because in the background they use the same web engine.

On other web sites it is never a problem to login in my firefox with Bitwarden.

If you need more please tell me.

**Reply by rijkvanzanten** [reply]
I think we'll have to consider this a bug in the web platform itself, rather than something we can fix in Directus. We just use a regular ol' input type="password" there, and like you mentioned already, it seems to work just fine with other browser/password manager combinations (i have no issues with 1Password mobile or desktop either) 🤔

**Reply by wendevlin** [reply]
> I think we'll have to consider this a bug in the web platform itself, rather than something we can fix in Directus. We just use a regular ol' input type="password" there, and like you mentioned already, it seems to work just fine with other browser/password manager combinations (i have no issues with 1Password mobile or desktop either) thinking

Thanks for testing. So on firefox for iOS login works with autofill from 1password?

I was afraid you going to say this. But since I never had any issue on other websites I am pretty sure it's on directus side.
The autofill itself works, so it's no issue in the html tag. It seems to be an issue the way the JS recognizes the input.

Furthermore I tested it with the iOS keychain and come to the same issue.
Moreover I tested it with my iPad 6.

**Reply by rijkvanzanten** [reply]
The login page is just a simple `<input type="password">`, there isn't any javascript magic watching that or responding to it, so I wouldn't really know what else to tweak for that particular combination of technologies :C

**Reply by wendevlin** [reply]
Hey. Ok now I got it ☹️ thanks for looking into it!
